### PR TITLE
fix: user's cursor can be offset for autocompletion

### PR DESCRIPTION
**Pull request by juleswritescode**
Suppose we have an incomplete query and the user's cursor is too far off:

`select * from           {}` 

In that case, we couldn't walk the tree-sitter tree.
When we tried to go to the child node at the cursor position, tree-sitter would stay on the `PROGRAM` node (every tree's head) which lead to unexpected contexts and infinite recursion.

That's now fixed! We simply "move the cursor" to the previous node:

`select * from       {}` becomes `select * from{}`.

Note:  Treesitter is already able to deal with ` {}        select * from`.

Some additional stuff:
- added a `tree-print` script to the `justfile` (that's much cooler than Makefile, btw)
- adjusted and used some test-helpers, adjusted the `tree_print` binary to ignore newlines

### CODE REVIEW

File: crates/pg_completions/src/context.rs [hunk 1]
```diff
@@ -127,9 +145,14 @@ impl<'a> CompletionContext<'a> {
...
             }
 
+            "keyword_from" => {
+                self.wrapping_clause_type = "keyword_from".try_into().ok();
+            }
+
             _ => {}
         }
 
+        // We have arrived at the leaf node
```
Comment (psteinroe) [inline crates/pg_completions/src/context.rs:155]: I love that you also use `we` when writing comments. it's so welcoming to read.

File: justfile [hunk 2]
```diff
@@ -130,6 +130,10 @@ new-crate name:
   cargo new --lib crates/{{snakecase(name)}}
   cargo run -p xtask_codegen -- new-crate --name={{snakecase(name)}}
 
+# Prints the treesitter tree of the given SQL file
+tree-print file:
+	cargo run --bin tree_print -- -f {{file}}
+
```
Comment (psteinroe) [inline justfile:136]: love it! will add this alter for the parser too

**Review by psteinroe — APPROVED** [review]
great work!!

File: crates/pg_completions/src/providers/tables.rs [hunk 3]
```diff
@@ -76,7 +76,7 @@ mod tests {
             (format!("select * from us{}", CURSOR_POS), "users"),
             (format!("select * from em{}", CURSOR_POS), "emails"),
             // TODO: Fix queries with tree-sitter errors.
```
Comment (juleswritescode) [inline crates/pg_completions/src/providers/tables.rs:78]:
```suggestion
```

**Pull request merged**